src/System.Net.Http/src/System/Net/Http/Managed/HttpEnvironmentProxy.cs

```diff
@@ -0,0 +1,258 @@
+                }
+            }
+        }
+
+        /// <summary>
+        /// This function will evaluate given string and it will try to convert
+        /// it to Uri object. The string could contain URI fragment, IP address and  port
+        /// tuple or just IP address or name. It will return null if parsing fails.
+        /// </summary>
+        private static Uri GetUriFromString(string value)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                return null;
+            }
+
+            if (!value.Contains("://"))
+            {
+                value = "http://" + value;
+            }
+
+            try
+            {
+                Uri uri = new Uri(value);
+                if (uri.Scheme == Uri.UriSchemeHttp || uri.Scheme == Uri.UriSchemeHttps)
+                {
+                    // We only support http and https for now.
+                    return uri;
+                }
+            }
+            catch { };
+
+            return null;
+        }
+
+        /// <summary>
+        /// This function returns true if given Host match bypass list.
+        /// Note, that the list is common for http and https.
+        /// </summary>
+        private bool IsMatchInBypassList(Uri input)
+        {
+            if (_bypass != null)
+            {
+                foreach (string s in _bypass)
+                {
+                    if (s[0] == '.')
+                    {
+                        // This should match either domain it self or any subdomain or host
+                        // .foo.com will match foo.com it self or *.foo.com
+                        if ((s.Length - 1) == input.Host.Length &&
+                            String.Compare(s, 1, input.Host, 0, input.Host.Length, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            return true;
+                        }
+                        else if (input.Host.EndsWith(s, StringComparison.OrdinalIgnoreCase))
+                        {
+                            return true;
+                        }
+
+                    }
+                    else
+                    {
+                        if (String.Equals(s, input.Host, StringComparison.OrdinalIgnoreCase))
+                        {
+                            return true;
+                        }
+                    }
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Gets the proxy URI. (iWebProxy interface)
+        /// </summary>
+        public Uri GetProxy(Uri uri)
+        {
+            return uri.Scheme == Uri.UriSchemeHttp ? _httpProxyUri : _httpsProxyUri;
+        }
+
+        /// <summary>
+        /// Checks if URI is subject to proxy or not.
+        /// </summary>
+        public bool IsBypassed(Uri uri)
+        {
+            bool ret =  (uri.Scheme == Uri.UriSchemeHttp ? _httpProxyUri : _httpsProxyUri) == null;
+
+            if (ret)
```

Comment:
This code pattern is weird, adding additional if checks, can be made to

```
return (uri.Scheme == Uri.UriSchemeHttp ? _httpProxyUri : _httpsProxyUri) == null ? true : IsMatchInBypassList(uri);
```

Reply:
agreed. I think we end up here after all the rewrites. I replaced it with 
return GetProxy(uri) == null ? true : IsMatchInBypassList(uri);
That should be pretty close to what you suggested.